source/main.cc

```diff
@@ -412,92 +434,71 @@ int main (int argc, char *argv[])
     {
       deallog.depth_console(0);
 
-      // print the basic header from processor 0
-      if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
-        print_aspect_header(std::cout);
+      int current_idx = 1;
+      const bool i_am_proc_0 = (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0);
 
+      std::string prm_name = "";
+      bool print_xml = false;
 
-      // check whether ASPECT was called with exactly one argument. if not,
-      // print an error message
-      //
-      // however, this does not work with PETSc because for PETSc, one
-      // may pass any number of flags on the command line; unfortunately,
-      // the PETSc initialization code (run through the call to
-      // MPI_InitFinalize above) does not filter these out, so all we
-      // can test is that there is *at least* one argument on the command
-      // line
-      if (
-        (
-#ifdef ASPECT_USE_PETSC
-          argc < 2
-#else
-          argc != 2
-#endif
-        )
-        // or if the user called aspect with either --help or -h as
-        // the sole arguments
-        ||
-        (
-          (
-#ifdef ASPECT_USE_PETSC
-            argc >= 2
-#else
-            argc == 2
-#endif
-          )
-          &&
-          (
-            (std::string(argv[1]) == "--help")
-            ||
-            (std::string(argv[1]) == "-h")
-          )
-        )
-      )
+      while (current_idx<argc)
```

Comment:
can you add a comment before this line of the form "Loop over all command line arguments. Handle a number of special ones starting with a dash, and then take the first non-special one as the name of the input file. We will later check that there are no further arguments left after that (though there may be with PETSc, see below)."